xen/drivers/passthrough/iommu.c

```diff
@@ -129,7 +129,7 @@ static void __init parse_iommu_param(char *s)
     } while ( ss );
 }
 
-int iommu_domain_init(struct domain *d)
+int iommu_domain_init(struct domain *d, bool_t use_iommu)
```

Comment:
This changes shouldn't be moved to arch_iommu_domain_init() since it is called before hd->platform_ops->init, but the iommu_construct() should be called after hd->platform_ops->init.